src/python/pants/backend/go/util_rules/binary.py

```diff
@@ -62,15 +66,24 @@ async def determine_main_pkg_for_go_binary(
             WrappedTarget,
             WrappedTargetRequest(specified_address, description_of_origin=description_of_origin),
         )
-        if not wrapped_specified_tgt.target.has_field(GoPackageSourcesField):
+        if not wrapped_specified_tgt.target.has_field(
+            GoPackageSourcesField
+        ) and not wrapped_specified_tgt.target.has_field(GoImportPathField):
             raise InvalidFieldException(
                 f"The {repr(GoBinaryMainPackageField.alias)} field in target {addr} must point to "
-                "a `go_package` target, but was the address for a "
+                "a `go_package` or `go_third_party_package` target, but was the address for a "
                 f"`{wrapped_specified_tgt.target.alias}` target.\n\n"
-                "Hint: you should normally not specify this field so that Pants will find the "
-                "`go_package` target for you."
+                "Hint: you should normally not specify this field for local packages so that Pants "
```

Comment:
s/local/first party?

Reply:
Hmm. I was trying to differentiate local being "in the directory the binary is defined in", and not just first-party in general - but maybe inference works cross-directory? If it doesn't, I don't think first-party is specific enough but open to rephrasing it in general.

Reply:
Inference only works in the current directory. Maybe be more explicit, like this:

> Hint: unless the package is a `go_third_party_package` target, then you should normally not ...

Reply:
Good idea; done!